### ISSUE TITLE
prefer-spread flagging `Array.from` for non-iterable objects

### ISSUE THREAD

**Issue by Nightmaster** [sindresorhus/eslint-plugin-unicorn#1154]
Same fundamental cause as reported in #1064, 
the slight difference here is it occurs while using `Array.from` with elements without `Symbol.iterator`, for example type: `NodeListOf`

Problematic rule is so `prefer-spread`

**Typescript**:
```typescript
 //  ↓ Raises "Prefer the spread operator over `Array.from(…)`" error
Array.from(document.querySelectorAll<HTMLDivElement>('div'))
[...document.querySelectorAll<HTMLDivElement>('div')] // ← Will not compile
```

**Comment by sindresorhus** [comment]
Duplicate of #687

**Comment by sindresorhus** [comment]
https://github.com/microsoft/TypeScript/issues/43225
